radicle-surf/src/vcs/git/commit.rs

```diff
@@ -166,3 +166,39 @@ impl<'repo> TryFrom<git2::Commit<'repo>> for Commit {
         })
     }
 }
+
+/// Supports various ways to specify a commit in Git.
+pub trait IntoCommit {
+    /// Resolves into a commit in `repo`.
+    fn peel(&self, repo: &RepositoryRef) -> Result<Commit, Error>;
+}
+
+impl IntoCommit for &Commit {
```

Comment:
I'd make the `clone` happen by the caller, not the trait.
```suggestion
impl IntoCommit for Commit {
```

Reply:
thanks.